### PR TITLE
feat: Refactor end-to-end test case creator script

### PR DESCRIPTION
**Pull request by ThomasGalesloot**
# Pull Request Key Information

## Reason for change

The End-to-End test case creator script was long, hard to maintain and did not share code with other scripts that "run" LDAR-Sim.

## What was changed

Refactored the end-to-end test case creator script to use an end-to-end test case creator simulation manager that inherits core functionality from the base simulation manager.

## Intended Purpose

This allows the end-to-end test case creator to inherit changes made to LDAR-Sim run functionality.

## Level of version change required

N/A

## Testing Completed

All unit tests pass:
[unit_test_results.txt](https://github.com/user-attachments/files/16516125/unit_test_results.txt)

All E2E tests pass:
[V4_simple_non_repairable_emissions_bb24139aabd09018a0ce99ce324c10cdab0c31b8.log](https://github.com/user-attachments/files/16516113/V4_simple_non_repairable_emissions_bb24139aabd09018a0ce99ce324c10cdab0c31b8.log)
[V4-simple_repairable_emissions_bb24139aabd09018a0ce99ce324c10cdab0c31b8.log](https://github.com/user-attachments/files/16516115/V4-simple_repairable_emissions_bb24139aabd09018a0ce99ce324c10cdab0c31b8.log)

Comparison of file names between master and this branch is identical (Actual file contents will not be because E2E creator script forces remake of generator files):
[E2E creator testing.zip](https://github.com/user-attachments/files/16516122/E2E.creator.testing.zip)

Sample E2E test case created using latest from script passes: 
[new_bb24139aabd09018a0ce99ce324c10cdab0c31b8.log](https://github.com/user-attachments/files/16516117/new_bb24139aabd09018a0ce99ce324c10cdab0c31b8.log)

## Target Issue

N/A

## Additional Information

N/A

### CODE REVIEW

File: LDAR_Sim/src/simulation/end_to_end_creation_simulation_manager.py
Comment (SEAJang) [inline LDAR_Sim/src/simulation/end_to_end_creation_simulation_manager.py:1]: Doc string please :(

Comment (SEAJang) [reply LDAR_Sim/src/simulation/end_to_end_creation_simulation_manager.py:1]: Nit: could we possibly move this file into the testing directory?

File: LDAR_Sim/src/simulation/end_to_end_creation_simulation_manager.py [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+            if item != ".gitignore":
+                if os.path.isfile(item_path):
+                    os.remove(item_path)
+                elif os.path.isdir(item_path):
+                    shutil.rmtree(item_path)
+
+    def retrieve_test_parameters(self) -> None:
+        self.original_inputs_dir: Path = self.root_dir / self.original_input_folder
+        self.original_params_dir: Path = self.root_dir / self.original_param_folder
+        self.params_dir: Path = self.test_creator_dir / "params"
```
Comment (SEAJang) [inline LDAR_Sim/src/simulation/end_to_end_creation_simulation_manager.py:62]: This I know is from the original, but do we have a constant that we can replace the "params" with?

File: LDAR_Sim/src/simulation/end_to_end_creation_simulation_manager.py [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+        self.sim_params: dict = input_manager.read_and_validate_parameters(parameter_filenames)
+        self.out_dir: Path = get_abs_path("./expected_outputs", self.test_case_dir)
+        self._set_parameters()
+
+    @override
+    def _set_parameters(self) -> None:
+        self.base_program: str = self.sim_params[pdc.Sim_Setting_Params.BASELINE]
+        self.in_dir = get_abs_path("./inputs", self.test_creator_dir)
+        shutil.copytree(self.original_inputs_dir, self.in_dir)
+        if os.path.exists(self.in_dir / "generator"):
```
Comment (SEAJang) [inline LDAR_Sim/src/simulation/end_to_end_creation_simulation_manager.py:95]: See above comment about "params"

File: LDAR_Sim/src/simulation/end_to_end_creation_simulation_manager.py [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+            shutil.rmtree(self.in_dir / "generator")
```
Comment (SEAJang) [inline LDAR_Sim/src/simulation/end_to_end_creation_simulation_manager.py:96]: See above comment

File: LDAR_Sim/src/simulation/end_to_end_creation_simulation_manager.py [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+        non_baseline_progs = get_non_baseline_prog_names(self.programs, self.base_program)
+        self.summary_stats_manager.gen_cost_summary_outputs(non_baseline_progs)
+
+    def format_test_results_location(self, tests_dir: Path) -> None:
+        os.chdir(self.root_dir)
+        shutil.move(self.params_dir, self.test_case_dir)
+        shutil.move(self.in_dir, self.test_case_dir)
+        shutil.move(self.test_case_dir, tests_dir / self.test_name)
+
+        test_generator = tests_dir / self.test_name / "inputs" / "generator"
```
Comment (SEAJang) [inline LDAR_Sim/src/simulation/end_to_end_creation_simulation_manager.py:123]: See above comment

File: LDAR_Sim/src/simulation/end_to_end_creation_simulation_manager.py [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+
+class EndToEndCreationSimulationManager(SimulationManager):
+    def __init__(
+        self,
+        input_manager: InputManager,
+        test_creator_dir: Path,
+        root_dir: Path,
+        script_arguments: list[str],
+    ):
+        self.GLOBAL_PARAMS_TO_REP: "dict[str, Any]" = {
```
Comment (SEAJang) [inline LDAR_Sim/src/simulation/end_to_end_creation_simulation_manager.py:24]: Nit: Could make this a class attribute

**Review by SEAJang — APPROVED** [review]
Great job

**Comment by ThomasGalesloot** [comment]
New E2E test results: 
[test_1db583486dfeb63dd0273ccbeda9773ab180d09d.log](https://github.com/user-attachments/files/16516548/test_1db583486dfeb63dd0273ccbeda9773ab180d09d.log)
[V4_simple_non_repairable_emissions_1db583486dfeb63dd0273ccbeda9773ab180d09d.log](https://github.com/user-attachments/files/16516549/V4_simple_non_repairable_emissions_1db583486dfeb63dd0273ccbeda9773ab180d09d.log)
[V4-simple_repairable_emissions_1db583486dfeb63dd0273ccbeda9773ab180d09d.log](https://github.com/user-attachments/files/16516550/V4-simple_repairable_emissions_1db583486dfeb63dd0273ccbeda9773ab180d09d.log)

**Pull request merged**